src/main/java/Verify.java

```diff
@@ -0,0 +1,104 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Checks all the possible errors that can occur during entry of command and
+ * throws DukeException with message
+ */
+public class Verify {
+    public Verify() {
+    }
+
+    /**
+     * 
+     * @param str the input string
+     * @return returns if the string is a number
+     */
+    public static boolean isNum(String str) {
+        try {
+            int i = Integer.parseInt(str);
+        } catch (NumberFormatException | NullPointerException nfe) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * 
+     * @param command   the string command entered by user
+     * @param list      the task list to compare with for checking validity
+     * @param formatter date formatter to catch wrong date format exceptions
+     * @throws DukeException in the case of invalid input
+     */
+    public static String checkCommandValidity(String command, TaskList list, DateTimeFormatter formatter)
```

Comment:
Would it be easier to test if this method were broken into smaller methods (e.g.  method to test the funtionality of todo method) and this method just called them using a switch case?

Reply:
yes, good idea. it would probably be simpler for both testing and reading if i split up the huge if-else block